### ISSUE TITLE
Custom preview breaks after update to 2.6.1

### ISSUE THREAD

**Issue by katcoutts** [netlify/netlify-cms#2230]
**Describe the bug**
I have a custom preview template for one of our collections and it was working fine in v 2.3 but I just updated to 2.6.1 and now it is fine for previewing completed blogs, but when you are creating one it breaks as soon as you enter a social description or title with an error telling me TypeError u is undefined.

**To Reproduce**
This is the preview code, which works perfectly for version 2.3 - as you create a blog it fills in each part.
```
var BlogPreview = createClass({
        getContent: function() {
          var entry = this.props.entry;
          var socialImage = this.props.widgetsFor('social').getIn(['data', 'image']);
          var tags = this.props.widgetsFor('tags').map(function(tag){
            if (tag !== undefined){
            return tag.getIn(['data']);
            }
            return "";
          });
          var previewContent = {
            title: entry.getIn(['data', 'title']) || "",
            body: this.props.widgetFor('body') || "",
            imageUrl: this.props.getAsset(socialImage),
            socialTitle: this.props.widgetsFor('social').getIn(['data', 'title']),
            socialDescription: this.props.widgetsFor('social').getIn(['data', 'description']),
            author: entry.getIn(['data', 'author']) || "None",
            tagsString: tags.join(', '),
            url: entry.getIn(['data', 'url_slug']) || "",
            metaDescription: entry.getIn(['data', 'description']) || "",
            publishedDate: entry.getIn(['data', 'date']) || "",
          };
          return previewContent;
        },
        render: function() {
          var previewContent = this.getContent();
          return h('div', {'className': 'u-background-color--fa-white'},
            h('div', {"className": "u-background-color--fa-white"},
            h('h1', {}, previewContent.title),
            h('div', {}, previewContent.body),
            h('div', {style: {'maxWidth': '80%', 'border' : '1px solid black'}, "className": "u-background-color--fa-blue--x-light u-margin-top--large"},
            h('strong', {}, 'URL: '),
            h('p', {}, previewContent.url ),
            h('strong', {}, 'Author: '),
            h('p', {}, previewContent.author),
            h('strong', {}, 'Page meta description: '),
            h('p', {}, previewContent.metaDescription),
            h('strong', {}, 'Published date: '),
            h('p', {}, previewContent.publishedDate),
            h('strong', {}, 'Tags:'),
            h('p', {}, previewContent.tagsString),
            h('strong', {}, 'Social and meta title: '),
            h('p', {}, previewContent.socialTitle),
            h('strong', {}, 'Social and blog index description:'),
            h('p', {}, previewContent.socialDescription),
            h('p', {},
            h('strong', {}, 'Social OG image:'),
            ),
            h('img', { src: previewContent.imageUrl.toString(), style: {'maxWidth': '50%'} }),
            ),
            ),
          );
        }
      });
```
**Expected behavior**
If I create a new Blog the preview should populate as I fill in the sections and not error.

**Screenshots**

**Applicable Versions:**

 - Netlify CMS version: 2.6.1
 - Git provider: GitHub
 - OS: Mojave
 - Browser version happening in all I've tried (safari, chrome, firefox)

 - Node.JS version:

**CMS configuration**
This is the config for the collection this relates to.
```yml
collections:
  - name: "blog"
    label: "Blog"
    folder: "source/blog"
    create: true
    extension: "html.md.erb"
    format: "frontmatter"
    slug: "{{year}}-{{url_slug}}"
    fields:
      - label: "URL"
        name: "url_slug"
        widget: "string"
        hint: "When creating a blog this will be the filename and therefore the url of your blog (eg. if you put tips-on-self-assessment it will live at blog/tips-on-self-assessment) - if you are trying to edit the filename of an existing blog you will need to do that via Github. Speak to the website team if you need help with this."
      - label: "Author"
        name: "author"
        widget: "select"
        options:
          - { label: "Name", value: "value" }
          - { label: "Another name", value: "something" }
        required: false
        hint: "Leave blank if there is no named author and it's not a guest author"
      - { label: "Title", name: "title", widget: "string", hint: "This is the title that will appear on the blog." }
      - { label: "Description", name: "description", widget: "string" }
      - label: "Published date"
        name: "date"
        widget: "datetime"
        format: "YYYY-MM-DD HH:mm"
      - label: "Tags"
        name: "tags"
        widget: "select"
        multiple: true
        options: ["Sample", "Example"]
      - label: "Social"
        name: "social"
        widget: "object"
        fields:
          - { label: "Title", name: "title", widget: "string" }
          - { label: "Description", name: "description", widget: "string" }
          - { label: "Image", name: "image", widget: "image", default: "/components/images/netlify/social-general.png" }
      - { label: "Body", name: "body", widget: "markdown" }
```

**Additional context**

**Comment by barthc** [comment]
@katcoutts could you please update the OP with contents of your CMS config ?

**Comment by katcoutts** [comment]
@barthc I've added the section of the config for this collection.

**Comment by barthc** [comment]
@erquhart  the culprit here is the `widget.js`  `onChangeObject` function. In this case it's passing `{social: undefined}` as `fieldsMetaData` value to the `widgetsFor` function. PR #2245

**Comment by katcoutts** [comment]
Thank you for picking this up

**Comment by katcoutts** [comment]
Hi. If I pull in the 2.8.1 beta version of netlifycms should this be working now? I tried it and my preview still gets an undefined error when you go to add the social info. 
![Screenshot 2019-03-27 at 15 31 50](https://user-images.githubusercontent.com/17859815/55089808-ed259580-50a5-11e9-8f13-a842f9b5de8f.png)

**Comment by barthc** [comment]
@katcoutts this fixed should be made available in the next release (not 2.8.1 beta)

**Comment by katcoutts** [comment]
Awesome. Thanks for clarifying!

**Comment by erquhart** [comment]
Fix released in 2.9.0.
